provider/pkg/await/deployment.go

```diff
@@ -150,34 +150,72 @@ func (dia *deploymentInitAwaiter) Await() error {
 		return err
 	}
 
-	// Create Deployment watcher.
-	deploymentWatcher, err := deploymentClient.Watch(context.TODO(), metav1.ListOptions{})
+	// Create Deployment watcher starting from the deployment's resource version, if set.
+	var depListOptions metav1.ListOptions
+	if dia.deployment != nil {
+		depListOptions.ResourceVersion = dia.deployment.GetResourceVersion()
+	}
+	deploymentWatcher, err := deploymentClient.Watch(context.TODO(), depListOptions)
```

Comment:
TODO: handle potential 410 - gone error response here: https://kubernetes.io/docs/reference/using-api/api-concepts/#410-gone-responses

Reply:
It seems like the fallback option could be grabbing the latest like it was doing before. [Condition 1](https://github.com/pulumi/pulumi-kubernetes/pull/1596/files#diff-db92f9606426da028402e0727166fdf10d9554932dad8ba121ef0448ccc08669R134-R136) is the only one that is comparing generations, and the generation just has to be later. If the specified resource version no longer exists, that implies that condition 1 is true.

Reply:
I have updated this to restart at the current latest and reset our references to the latest deployment state.